tiny_dnn/activations/softmax_layer.h

```diff
@@ -0,0 +1,58 @@
+/*
+    Copyright (c) 2017, Taiga Nomi
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+#include "tiny_dnn/activations/activation_layer.h"
+#include "tiny_dnn/layers/layer.h"
+
+namespace tiny_dnn {
+
+class softmax_layer : public activation_layer {
+ public:
+  using activation_layer::activation_layer;
+
+  std::string layer_type() const override { return "softmax-activation"; }
+
+  void forward_activation(const vec_t &x, vec_t &y) override {
+    float_t alpha = *std::max_element(x.begin(), x.end());
+    float_t denominator(0);
+    for (serial_size_t j = 0; j < x.size(); j++) {
+      y[j] = std::exp(x[j] - alpha);
+      denominator += y[j];
+    }
+
+    float_t recip_denominator = float_t(1) / denominator;
+    for (serial_size_t j = 0; j < x.size(); j++) {
+      y[j] *= recip_denominator;
+    }
+  }
+
+  void backward_activation(const vec_t &x,
+                           const vec_t &y,
+                           vec_t &dx,
+                           const vec_t &dy) override {
+    const serial_size_t len = static_cast<serial_size_t>(dy.size());
+    float_t yj;
+
+    // auxilliary vector to store element wise softmax gradients of all elements
+    thread_local vec_t df;
```

Comment:
@beru there's no plan to drop support for OSX and Windows. Actually, I think it's a valuable feature of this library. Please, don't confuse possible external readers. If you think the contrary, open a ticket so that we can discuss there.